test/tests/functional/pbs_power_provisioning_cray.py

```diff
@@ -231,28 +263,128 @@ def test_cray_eoe_job(self):
+        """
+        Submit jobs requesting multiple eoe and job should rejected by qsub.
+        """
+
+        self.setup_cray_eoe()   # setup hook and eoe
+        self.enable_power()             # enable hooks
+
+        a = {'Resource_List.place': 'scatter',
+             'Resource_List.select': '10:eoe=low+10:eoe=high'}
+        j = Job(TEST_USER, attrs=a)
+        j.set_sleep_time(10)
+        try:
+            self.server.submit(j)
+        except PbsSubmitError as e:
+            self.assertTrue(
+                'Invalid provisioning request in chunk' in e.msg[0])
+
+    @timeout(700)
+    def test_cray_server_prov_off(self):
+        """
+        Submit jobs requesting eoe when power provisioning unset on server
+        and verify that jobs wont run.
+        """
+
+        self.setup_cray_eoe()   # setup hook and eoe
+        self.enable_power()             # enable hooks
+        eoes = set(['low', 'med', 'high'])
+        a = {'power_provisioning': 'False'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        while len(eoes) > 0:
+            eoe = eoes.pop()
+            jid = self.submit_job(10,
+                                  {'Resource_List.place': 'scatter',
+                                   'Resource_List.select': '%d:eoe=%s'
+                                   % (self.npp, eoe)})
+            self.server.expect(JOB, {
+                'job_state': 'Q',
+                'comment': 'Not Running: No available resources on nodes'},
+                id=jid)
+
+    @timeout(700)
+    def test_cray_node_prov_off(self):
+        """
+        Submit jobs requesting eoe and verify that jobs wont run on
+        nodes where power provisioning is set to false.
+        """
+
+        self.setup_cray_eoe()   # setup hook and eoe
+        self.enable_power()             # enable hooks
+        eoes = set(['low', 'med', 'high'])
+        # set power_provisioning to off where eoe is set to false
+        for i in range(0, self.npp):
+            a = {'power_provisioning': 'False'}
+            self.server.manager(MGR_CMD_SET, NODE, a, id=self.names[i])
+
+        while len(eoes) > 1:
+            eoe = eoes.pop()
+            jid = self.submit_job(10,
+                                  {'Resource_List.place': 'scatter',
+                                   'Resource_List.select': '%d:eoe=%s'
+                                   % (self.npp, eoe)})
+            self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        jid_low = self.submit_job(10,
+                                  {'Resource_List.place': 'scatter',
+                                   'Resource_List.select': '%d:eoe=%s'
+                                   % (self.npp, 'low')})
+        exp_comm = 'Not Running: Insufficient amount of resource: '
+        exp_comm += 'vntype (cray_compute != cray_login)'
+        self.server.expect(JOB, {
+                           'job_state': 'Q',
+                           'comment': exp_comm}, id=jid_low)
```

Comment:
expect() does an or match by default.  You need to say op=PTL_AND if you want to do do an and match.  Either that or use two expects